### ISSUE TITLE
Change default name case for event names to match documentation

### ISSUE THREAD

**Issue by posva** [vuejs/eslint-plugin-vue#1759]
**What rule do you want to change?**

https://eslint.vuejs.org/rules/custom-event-name-casing.html#camelcase

**Does this change cause the rule to produce more or fewer warnings?**

Same

**How will the change be implemented? (New option, new default behavior, etc.)?**

Default change

**Please provide some example code that this change will affect:**

```js
      this.$emit('my-event')
      this.$emit('update:myProp', myProp)

      this.$emit('myEvent')
```

**What does the rule currently do for this code?**

Allows the 2 first lines, fails on the last.

**What will the rule do after it's changed?**

It would do the opposite.

It should inverse the default and enforce camel-cased event names when calling `emit()` like [proposed in docs](https://vuejs.org/guide/components/events.html#emitting-and-listening-to-events)

**Additional context**

**Comment by ota-meshi** [comment]
Thank you for posting this issue.
I didn't know the documentation was updated. I think we need to change the default option for the rule in the next major version.

**Comment by FloEdelmann** [comment]
For reference, the relevant section in the docs: https://vuejs.org/guide/components/events.html#emitting-and-listening-to-events
